Filebeat panic: name gzip_started already used

This seems to be some sort of race condition that so far I only managed to reproduce when running `TestFilebeatOTelMultipleReceiversE2E`. Use of otel receivers in Filebeat might also be affected.

The code generating the panic is not released, nor is it in a BC.

**Steps to reproduce**
1. cd `x-pack/filebeat`
1. Build the system test binary with otel support (PR coming soon). Or edit the `magefile.go` so `BuildSystemTestBinary` calls `devtools.BuildSystemTestOTelBinary()`.
2. Run the test a few times `go test -tags=integration -v -run=TestFilebeatOTelMultipleReceiversE2E ./tests/integration`

Some times it will fail because of https://github.com/elastic/beats/issues/45631, others it will fail with a panic:
```
        panic: name gzip_started already used
        
        goroutine 266 [running]:
        github.com/elastic/elastic-agent-libs/monitoring.panicErr(...)
                /home/tiago/go/pkg/mod/github.com/elastic/elastic-agent-libs@v0.21.4/monitoring/registry.go:287
        github.com/elastic/elastic-agent-libs/monitoring.(*Registry).doAdd(0xc001b54980, {0xebaae71?, 0x0?}, {0xfb4e700, 0xc003470a90}, 0xc0023c4f48)
                /home/tiago/go/pkg/mod/github.com/elastic/elastic-agent-libs@v0.21.4/monitoring/registry.go:189 +0x7d
        github.com/elastic/elastic-agent-libs/monitoring.addVar(0xc001b54980, {0xebaae71, 0xc}, {0x0?, 0xc00227da10?, 0xc00250d7a0?}, {0xfb4e700, 0xc003470a90}, {0xfb4e5c0, 0xc001d40910})
                /home/tiago/go/pkg/mod/github.com/elastic/elastic-agent-libs@v0.21.4/monitoring/metrics.go:263 +0x74
        github.com/elastic/elastic-agent-libs/monitoring.NewInt(0x15acf2e0?, {0xebaae71, 0xc}, {0x0, 0x0, 0x0})
                /home/tiago/go/pkg/mod/github.com/elastic/elastic-agent-libs@v0.21.4/monitoring/metrics.go:57 +0x19c
        github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile.NewMetrics(0xc0025cff00)
                /home/tiago/devel/beats/filebeat/input/filestream/internal/input-logfile/metrics.go:98 +0x4bd
        github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile.(*managedInput).Run(_, {0xc003510cf0, {0xc0022bf698, 0x15}, {0xc0022bf698, 0x15}, {0xeb98f8d, 0xa}, {{0xeb8dc60, 0x8}, ...}, ...}, ...)
                /home/tiago/devel/beats/filebeat/input/filestream/internal/input-logfile/input.go:67 +0x1ae
        github.com/elastic/beats/v7/filebeat/input/v2/compat.(*runner).Start.func1()
                /home/tiago/devel/beats/filebeat/input/v2/compat/compat.go:162 +0x2c8
        created by github.com/elastic/beats/v7/filebeat/input/v2/compat.(*runner).Start in goroutine 91
                /home/tiago/devel/beats/filebeat/input/v2/compat/compat.go:139 +0xb9
    framework.go:1052: Last 0 bytes of stdout:
    framework.go:740: Temporary directory saved: /home/tiago/devel/beats/x-pack/filebeat/build/integration-tests/TestFilebeatOTelMultipleReceiversE2E304441384
--- FAIL: TestFilebeatOTelMultipleReceiversE2E (30.03s)
FAIL
FAIL    github.com/elastic/beats/v7/x-pack/filebeat/tests/integration   30.049s
FAIL
```

The panic is coming from
https://github.com/elastic/beats/blob/7fbc298248653d9542fb496a580f4d612cb779ed/filebeat/input/filestream/internal/input-logfile/metrics.go#L65-L109

Technically any of the metrics could cause the panic, but on my tests it has always been `gzip_started`: https://github.com/elastic/beats/blob/7fbc298248653d9542fb496a580f4d612cb779ed/filebeat/input/filestream/internal/input-logfile/metrics.go#L98

--- a/CHANGELOG.next.asciidoc
+++ b/CHANGELOG.next.asciidoc
@@ -21,7 +21,6 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - The Beats logger and file output rotate files when necessary. The beat now forces a file rotation when unexpectedly writing to a file through a symbolic link.
 - Allow faccessat(2) in seccomp. {pull}43322[43322]
 - Update user agent used by beats http clients {pull}45251[45251]
-- Fix a race condition during metrics initialization which could cause a panic. {issue}45822[45822] {pull}45967[45967]
 
 *Auditbeat*
 
@@ -150,6 +149,7 @@ otherwise no tag is added. {issue}42208[42208] {pull}42403[42403]
 - Use Debian 11 to build linux/arm to match linux/amd64. Upgrades linux/arm64's statically linked glibc from 2.28 to 2.31. {issue}44816[44816]
 - The Elasticsearch output now correctly applies exponential backoff when being throttled by 429s ("too many requests") from Elasticsarch. {issue}36926[36926] {pull}45073[45073]
 - Fixed case where Beats would silently fail due to invalid input configuration, now the error is correctly reported. {issue}43118[43118] {pull}45733[45733]
+- Fix a race condition during metrics initialization which could cause a panic. {issue}45822[45822] {pull}46054[46054]
 
 *Auditbeat*
 

